### PR TITLE
Add ipxe support for arm64

### PR DESCRIPTION
**Pull request by jeffdyoung**
Waiting for ipxe artifacts to become available.

https://issues.redhat.com/browse/ARMOCP-236

### CODE REVIEW

**Comment by yselkowitz** [comment]
The main-packages-list files need to be adjusted for the forthcoming packages names.

**Comment by yselkowitz** [comment]
Filed https://bugzilla.redhat.com/show_bug.cgi?id=2073842 for the new mtools error; a workaround is:

```diff
--- a/prepare-efi.sh
+++ b/prepare-efi.sh
@@ -23,1 +23,1 @@ else
-dnf install -y grub2 shim dosfstools mtools $PACKAGES
+dnf install -y glibc-gconv-extra grub2 shim dosfstools mtools $PACKAGES
```

**Comment by elfosardo** [comment]
> Filed https://bugzilla.redhat.com/show_bug.cgi?id=2073842 for the new mtools error; a workaround is:
> 
> ```diff
> --- a/prepare-efi.sh
> +++ b/prepare-efi.sh
> @@ -23,1 +23,1 @@ else
> -dnf install -y grub2 shim dosfstools mtools $PACKAGES
> +dnf install -y grub2 shim dosfstools mtools glibc-gconv-extra $PACKAGES
> ```

@yaacov thanks for opening the BZ, this has been already worked around upstream https://github.com/metal3-io/ironic-image/blob/main/Dockerfile#L9
I'm going to move the glibc-gconv-extra to prepare-efi.sh so we don't diverge too much

**Comment by elfosardo** [comment]
/hold
this should wait for https://github.com/openshift/ironic-image/pull/267

File: ironic-config/dnsmasq.conf.j2 [hunk 1]
```diff
@@ -23,10 +23,13 @@ dhcp-boot=tag:ipxe,http://{{ env.IRONIC_URL_HOST }}:{{ env.HTTP_PORT }}/boot.ipx
 # Note: Need to test EFI booting
 dhcp-match=set:efi,option:client-arch,7
 dhcp-match=set:efi,option:client-arch,9
-dhcp-match=set:efi,option:client-arch,11
-# Client is PXE booting over EFI without iPXE ROM; send EFI version of iPXE chainloader
+# Client is PXE booting over EFI without iPXE ROM; send EFI version of iPXE chainloader for x86
 dhcp-boot=tag:efi,tag:!ipxe,snponly.efi
 
+# Client is PXE booting over EFI without iPXE ROM; send EFI version of iPXE chainloader for aarch64
+dhcp-match=set:efi,option:client-arch,11
```
Comment (elfosardo) [inline ironic-config/dnsmasq.conf.j2:30]: let's move this back to L26 but change the set to efi_arm64

File: ironic-config/dnsmasq.conf.j2 [hunk 2]
```diff
@@ -23,10 +23,13 @@ dhcp-boot=tag:ipxe,http://{{ env.IRONIC_URL_HOST }}:{{ env.HTTP_PORT }}/boot.ipx
...
+dhcp-boot=tag:efi,tag:!ipxe,/arm64-efi/snponly.efi
```
Comment (elfosardo) [inline ironic-config/dnsmasq.conf.j2:31]: let's set the tag here to efi_arm64

**Review by elfosardo — CHANGES_REQUESTED** [review]
the errors seem related to this change, please check my comments

**Comment by yselkowitz** [comment]
How do we need to handle upstreaming this to metal3 so that these changes don't get lost at the next sync?  The upstream Dockerfile builds ipxe from a git checkout which is newer than the C9S RPM packages, but that doesn't really give a way to build both x86 and arm simultaneously.  Does that mean the additions to `scripts/rundnsmasq` need to be made conditional?  (prepare-ipxe.sh is downstream-specific so no changes needed there.)  Are the changes to dnsmasq.conf.j2 safe even if the arm64 efi files are absent upstream?

**Comment by elfosardo** [comment]
> How do we need to handle upstreaming this to metal3 so that these changes don't get lost at the next sync? The upstream Dockerfile builds ipxe from a git checkout which is newer than the C9S RPM packages, but that doesn't really give a way to build both x86 and arm simultaneously. Does that mean the additions to `scripts/rundnsmasq` need to be made conditional? (prepare-ipxe.sh is downstream-specific so no changes needed there.) Are the changes to dnsmasq.conf.j2 safe even if the arm64 efi files are absent upstream?

the upstream-downstream sync is done manually so we preserve what's needed for downstream
I don't think we need to make the changes conditional, but I guess proposing the change upstream and see how the CI reacts would be a good test

**Review by elfosardo — APPROVED** [review]

**Pull request merged**